hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileScanner.java

```diff
@@ -135,11 +136,17 @@ public static List<StoreFileScanner> getScannersForStoreFiles(Collection<HStoreF
     for (HStoreFile file : files) {
       // The sort function needs metadata so we need to open reader first before sorting the list.
       file.initReader();
```

Comment:
I wonder if we could have a DualStoreFileManager as well. That way, we could already keep separate list of store files there and it would contain the logic about which files should be returned for the scanner. It would also avoid us from having to open a reader on files we may not be interested.

Reply:
Good suggestion! I will add DualStoreFileManager.